### PR TITLE
remove starcoin_dao.rs, update starcoin-framework, update halley genesis

### PR DESCRIPTION
**Pull request by nkysg**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

File: executor/tests/module_upgrade_test.rs [hunk 1]
```diff
@@ -290,91 +283,6 @@ fn test_stdlib_upgrade_before_v12() -> Result<()> {
     Ok(())
 }
 
-#[stest::test(timeout = 3000)]
-fn test_stdlib_upgrade_since_v12() -> Result<()> {
```
Comment (sanlee42) [inline executor/tests/module_upgrade_test.rs:294]: use #[ignore] to do it.

File: executor/tests/sip_flag_test.rs [hunk 2]
```diff
@@ -1,16 +1,16 @@
 // Copyright (c) The Starcoin Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 use anyhow::Result;
-use starcoin_crypto::hash::PlainCryptoHash;
+// use starcoin_crypto::hash::PlainCryptoHash;
```
Comment (sanlee42) [inline executor/tests/sip_flag_test.rs:4]: Remove comment codes

Comment (nkysg) [reply executor/tests/sip_flag_test.rs:4]: get it

File: testsuite/features/cmd.feature [hunk 3]
```diff
@@ -186,74 +186,74 @@ Feature: cmd integration test
...
-    Then cmd: "account unlock"
-    Then cmd: "account execute-function --function 0x1::Block::checkpoint_entry -b"
-    Then cmd: "dev call-api chain.get_block_by_number [1,{\"raw\":true}]"
-    Then cmd: "account execute-function --function 0x1::Block::update_state_root_entry --arg {{$.dev[1].ok.raw.header}} -b"
-    Then cmd: "dev call --function 0x1::Block::latest_state_root"
-    Then assert: "{{$.dev[2].ok[1]}} == {{$.dev[1].ok.header.state_root}}"
-
-    Examples:
-      |  |
+#  Scenario Outline: [cmd] starcoin-framework checkpoint
```
Comment (sanlee42) [inline testsuite/features/cmd.feature:189]: Replace [cmd] with [ignore] to diable the testsuite

File: testsuite/features/cmd.feature [hunk 4]
```diff
@@ -186,74 +186,74 @@ Feature: cmd integration test
...
+#    Then cmd: "account execute-function --function 0x1::Block::update_state_root_entry --arg {{$.dev[1].ok.raw.header}} -b"
+#    Then cmd: "dev call --function 0x1::Block::latest_state_root"
+#    Then assert: "{{$.dev[2].ok[1]}} == {{$.dev[1].ok.header.state_root}}"
+#
+#    Examples:
+#      |  |
 
 
   #easy gas testing
-  Scenario Outline: [cmd] starcoin easy gas test
```
Comment (sanlee42) [inline testsuite/features/cmd.feature:203]: Same as above.

File: executor/tests/module_upgrade_test.rs (hunk 1 above)
Comment (nkysg) [reply executor/tests/module_upgrade_test.rs:294]: get it

File: testsuite/features/cmd.feature (hunk 3 above)
Comment (nkysg) [reply testsuite/features/cmd.feature]: get it

File: testsuite/features/cmd.feature (hunk 4 above)
Comment (nkysg) [reply testsuite/features/cmd.feature:203]: get it

**Review by sanlee42 — APPROVED** [review]

**Pull request merged**